cachix-api/src/Cachix/Api.hs

```diff
@@ -106,13 +109,21 @@ data BinaryCacheStreamingAPI route
           :: route
                :- CachixAuth
                :> "nar"
-               :> Capture "nar" NarC
+               :> Capture "nar" NarFileName
                :> StreamGet NoFraming OctetStream (ConduitT () ByteString (ResourceT IO) ()),
         createNar
           :: route
                :- "nar"
                :> StreamBody NoFraming XNixNar (ConduitT () ByteString (ResourceT IO) ())
-               :> Post '[JSON] NoContent
+               :> Post '[JSON] NoContent,
+        serveNar 
+          :: route
+              :- CachixAuth
+              :> "serve"
+              :> Capture "storepath" Text
```

Comment:
Unless the implementation changed since the demo link I saw, this is actually a store hash including `.narinfo`. Will it support proper store paths?

Reply:
It changed, it's now only a store hash. It used to be `NarInfoC`, now it's just `Text`.

Reply:
Cool. If we also strip off the `-$name` we can serve working hyperlinks to store paths (!)